extensions/mapreduce/src/main/java/org/radargun/stages/mapreduce/MapReduceStage.java

```diff
@@ -121,20 +126,31 @@
    @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
    private MapReducer<KOut, VOut, R> mapReducer;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Clustered clustered;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private CacheInformation cacheInformation;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private BasicOperations basicOperations;
 
    @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Iterable iterable;
 
    private boolean supportsResultCacheName = false;
 
+   @Init
+   public void validate() {
```

Comment:
Spark does not have the cache, so it should be separable. Why do you think you can't separate those? It seems to me that the code dealing with cache here should just do what Infinispan would do internally if you some option was supported; why can't that be moved to another stage?
Anyway, I would really love to see a single benchmark that would do the same MR job in both Infinispan and Spark. And I mean the same job, not different implementations of WordCount.